runtime/commonMain/src/kotlinx/serialization/internal/Enums.kt

```diff
@@ -5,13 +5,17 @@
 package kotlinx.serialization.internal
 
 import kotlinx.serialization.*
+import kotlin.jvm.JvmOverloads
 import kotlin.reflect.KClass
 
-public class EnumDescriptor(override val name: String, private val choices: Array<String>) : SerialClassDescImpl(name) {
+internal class EnumDescriptor @JvmOverloads constructor(
+    override val name: String,
+    private val values: Array<String> = emptyArray()
+) : SerialClassDescImpl(name) {
```

Comment:
Now enums support `SerialInfo`, it's worth to add some tests

Reply:
`SerialName/SerialId` saving is tested here: https://github.com/kotlin/kotlinx.serialization/blob/c95ab9846a5df82ffb98c26d782891d4ddce2fc0/runtime/commonTest/src/kotlinx/serialization/kinds/SerializableEnumTest.kt#L45. Do you think I've missed something?